aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -85,26 +86,33 @@ class DispatchTable final {
    * Remove the catch-all kernel.
    */
   void removeCatchallKernel() {
-    TORCH_INTERNAL_ASSERT(catchallKernel_.has_value(), "Tried to remove the catch-all kernel for operator ", operatorName_," but there is no catch-all kernel registered.");
-    catchallKernel_ = c10::nullopt;
+    TORCH_INTERNAL_ASSERT(catchallKernel_.isValid(), "Tried to remove the catch-all kernel for operator ", operatorName_," but there is no catch-all kernel registered.");
+    catchallKernel_ = {};
   }
 
   bool isEmpty() const {
-   return !catchallKernel_.has_value() && kernels_.size() == 0;
+   return !catchallKernel_.isValid() && kernelCount_ == 0;
   }
 
   std::string listAllDispatchKeys() const {
     std::ostringstream str;
     str << "[";
 
-    if (kernels_.size() != 0) {
-      str << toString(kernels_.begin()->first);
-      for (auto iter = ++kernels_.begin(); iter != kernels_.end(); ++iter) {
-        str << ", " << toString(iter->first);
+    if (kernelCount_ != 0) {
```

Comment:
Since this is the only place where we need `kernelCount_`, it might be better to not store that in an extra field (allowing for a potential invariant violation between `kernelCount_` and the content of `kernels_`), but instead rewrite the logic here to not need `kernelCount_`.